app/controllers/admin/consent_grants_controller.rb

```diff
@@ -3,10 +3,30 @@ class ConsentGrantsController < AdminController
     load_and_authorize_resource
 
     def index
-      @consent_grants = @consent_grants.by_date
+      @consent_grants = find_consent_grants
     end
 
     def show
     end
+
+    private
+
+    def find_consent_grants
+      if params[:search].present?
+        search = params[:search]
+        if search["reference"].present?
+          return ConsentGrant.where(guid: search["reference"]).by_date
+        end
+        if search["mpxn"].present?
+          meter = Meter.find_by_mpan_mprn(search["mpxn"])
+          if meter.present?
+            return ConsentGrant.where(school: meter.school).by_date
+          else
+            return []
+          end
+        end
+      end
+      return ConsentGrant.by_date
```

Comment:
"return" statements at end of methods generally unnecessary and frowned upon in idiomatic Ruby..! :-)